y/y.go

```diff
@@ -267,6 +271,9 @@ func NewThrottle(max int) *Throttle {
 // are already maximum number of workers working. If it detects an error from
 // previously Done workers, it would return it.
 func (t *Throttle) Do() error {
+	if t.finished {
```

Comment:
@ekalinin, consider a scenario where Do() and Finish() are called by different goroutines. In that case, there is a race condition. 
Thread 1 passes this check while thread 2 closes the channel and then thread 1 push into closed channel `ch`.
Try running this scenario using --race flag.